app/src/main/java/de/danoeh/antennapod/fragment/CoverFragment.java

```diff
@@ -184,20 +185,35 @@ public float convertDpToPixel(float dp) {
 
     private void configureForOrientation(Configuration newConfig) {
         LinearLayout mainContainer = getView().findViewById(R.id.cover_fragment);
-        ViewGroup.LayoutParams params = imgvCover.getLayoutParams();
+        LinearLayout textContainer = getView().findViewById(R.id.cover_fragment_text_container);
+
+        LinearLayout.LayoutParams params = (LinearLayout.LayoutParams) imgvCover.getLayoutParams();
+        LinearLayout.LayoutParams textParams = (LinearLayout.LayoutParams) textContainer.getLayoutParams();
+
 
         if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
+            double ratio = (float) newConfig.screenHeightDp / (float) newConfig.screenWidthDp;
+            double percentageWidth = 0.8;
+            if (ratio <= SIXTEEN_BY_NINE) {
+                percentageWidth = (ratio / SIXTEEN_BY_NINE) * percentageWidth * 0.8;
+            }
             mainContainer.setOrientation(LinearLayout.VERTICAL);
             if (newConfig.screenWidthDp > 0) {
-                params.width = (int) (convertDpToPixel(newConfig.screenWidthDp) * .80);
+                params.width = (int) (convertDpToPixel(newConfig.screenWidthDp) * percentageWidth);
                 params.height = params.width;
+                textParams.weight = 0;
                 imgvCover.setLayoutParams(params);
             }
         } else {
+            double ratio = (float) newConfig.screenHeightDp / (float) newConfig.screenWidthDp;
```

Comment:
Calculating the ratio can be moved out of the `if` statement to reduce code duplication

(sorry for the duplicate email)